t51-projects/gh-project-board.php

```diff
@@ -1,45 +1,70 @@
 <?php
 
 
-function curl_fetch() {
+$url = 'https://api.github.com/users/jessLundie'; // Set source URL
 
-	$url = 'https://api.github.com/users/jessLundie'; // Set source URL
+$media_header = array(
+	"Accept: application/vnd.github.inertia-preview+json",
+); // Custom media header required for projects in beta
 
-	$column_id = "15445284"; // Column id, found in the column link here: https://d.pr/i/W7RQJB
+$user_agent = "jessLundie"; // Set user agent
 
-	$project_url = "https://api.github.com/projects/columns/" . $column_id . "/cards"; // Set project URL
+$token = ""; // Basic Auth Token
 
-	$token = "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"; // Basic Auth Token
+$column_id = "15445284"; // Column id, found in the column link here: https://d.pr/i/W7RQJB
 
-	$user_agent = "jessLundie"; // Set user agent
+$project_url = "https://api.github.com/projects/columns/" . $column_id . "/cards"; // Set API URL for project
 
-	// Custom media header required for projects in beta
-	$headers = array(
-		"Accept: application/vnd.github.inertia-preview+json",
-	);
+function curl_fetch( $url, $media_header, $user_agent, $token, $api_url ) {
 
 	// Create a new cURL resource
 	$process = curl_init($url);
 
 	// Set options
-	curl_setopt( $process, CURLOPT_HTTPHEADER, $headers );
+	curl_setopt( $process, CURLOPT_HTTPHEADER, $media_header );
 	curl_setopt( $process, CURLOPT_USERAGENT, $user_agent );
 	curl_setopt( $process, CURLOPT_USERPWD, "$user_agent:$token" );
 	curl_setopt( $process, CURLOPT_RETURNTRANSFER, TRUE);
-	curl_setopt( $process, CURLOPT_URL, $project_url );
+	curl_setopt( $process, CURLOPT_URL, $api_url );
 
 	// Get project details
 	$result = curl_exec( $process );
 
 	// Close cURL
 	curl_close( $process );
 
+	$result = json_decode( $result, true ); // Convert response to array
```

Comment:
here's a small note about comments.

comments are _always great_ right?
well, comments, like code, have different value and there are different approaches to leaving them.

what I think happened here is that you were documenting your _process_ of understanding the problem, of how you came to solve it.

this is fun and tells a good story, but there's something in this comment that's _confusing_. this comment is _misleading_ because it asserts some behavior not obvious, not even inferable, from the code itself.

this lines absolutely _does not_ "convert [the] response to [an] array." it's possible that this could happen, if the response is the right shape, but other things can happen too given the same code and a different response.

```php
json_decode( "true", true ) === true;
json_decode( "3.141", true ) === 3.141;
json_decode( "null", true ) === NULL
```

These counter-examples contradict the comment and that's what I wanted to pause on to reflect. I love sharing the _why_ of code or the surprises of what's going on as a way to help someone new orient themselves here, but we can do better with this comment.

In a simple case like this nobody's going to go astray, but it's a common-enough type of situation to be a learning moment. Let's make a few observations:

 - the _code_ itself shows that what we are doing is decoding a JSON string. we might have to know a little about JSON and a little about `json_decode()` to know what's going on, but those are parts of the base knowledge we can assume people have to learn anyway.
 - "array" means different things in PHP than in JavaScript/JSON. in JSON it's always a list of things but in PHP it's _either_ a list _or_ a "map" (an associative array).
 - we're clearly expecting a JSON _map_ in the response.

A way we can change this comment is:
 - take it out since the code tells "what" we are doing
 - explain "why" we are doing it with something like ` // the response should be a JSON map`
 - explain fully in code with an additional constraint/check

If we want to make it clear what we're doing and we don't want to worry about the check we can add more context and safety with runtime assertions. This is probably more than we need to worry about here.

Reply:
Thank you! This makes so much sense.

``` 
If we want to make it clear what we're doing and we don't want to worry about the check we can add more context and safety with runtime assertions. 
```

Could you clarify what you mean here by runtime assertions? Or is that the bit that's more than we need to worry about here?